mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -0,0 +1,116 @@
+from mlflow.store.tracking.abstract_store import AbstractStore
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, RESOURCE_ALREADY_EXISTS, \
+    INVALID_STATE, RESOURCE_DOES_NOT_EXIST, INTERNAL_ERROR
+from mlflow.entities import (
+    Experiment, RunTag, Metric, Param, RunData, RunInfo,
+    SourceType, RunStatus, Run, ViewType, ExperimentTag, Columns, LifecycleStage)
+from mlflow.exceptions import MlflowException
+import uuid
+from mlflow.utils.uri import append_to_uri_path
+from elasticsearch_dsl import Index, Search, Q, connections
+from models import ElasticExperiment, ElasticRun, ElasticMetric, \
+    ElasticParam, ElasticTag, ElasticLatestMetric
+
+user = "a.vivien"
+password = "Akuseru2509"
+
+url = user + ":" + password + "@10.236.61.26:80"
+
+
+class ElasticsearchStore(AbstractStore):
+
+    ARTIFACTS_FOLDER_NAME = "artifacts"
+    DEFAULT_EXPERIMENT_ID = "0"
+
+    def __init__(self):
+        connections.create_connection(hosts=[url])
+
+    def create_experiment(self, name, artifact_location=None):
+        if name is None or name == '':
+            raise MlflowException('Invalid experiment name', INVALID_PARAMETER_VALUE)
+        experiment = ElasticExperiment(name=name, lifecycle_stage=LifecycleStage.ACTIVE,
+                                       artifact_location=artifact_location)
+        experiment.save()
+        return str(experiment.meta.id)
+
+    def get_experiment(self, experiment_id):
+        experiment = ElasticExperiment.get(id=experiment_id)
+        return experiment
+
+    def create_run(self, experiment_id, user_id, start_time, tags):
+        run_id = uuid.uuid4().hex
+        # experiment = self.get_experiment(experiment_id)
+        # artifact_location = append_to_uri_path(experiment.artifact_location, run_id,
+        #                                       ElasticsearchStore.ARTIFACTS_FOLDER_NAME)
+        artifact_location = "artifact_path"
+
+        tags_dict = {}
+        for tag in tags:
+            tags_dict[tag.key] = tag.value
+        run_tags = [ElasticTag(key=key, value=value) for key, value in tags_dict.items()]
+        run = ElasticRun(meta={'id': run_id},
+                         experiment_id=experiment_id, user_id=user_id,
+                         status=RunStatus.to_string(RunStatus.RUNNING),
+                         start_time=start_time, end_time=None,
+                         lifecycle_stage=LifecycleStage.ACTIVE, artifact_uri=artifact_location,
+                         tags=run_tags)
+        run.save()
+        return run.to_mlflow_entity()
+
+    def get_run(self, run_id):
+        run = self._get_run(run_id=run_id)
+        return run.to_mlflow_entity()
+
+    def _get_run(self, run_id):
+        run = ElasticRun.get(id=run_id)
+        return run
```

Comment:
Seeing the code, it is useless to have a private _get_run method, you can call directly the ElasticRun.get in the other get_run